src/components/Admin/AdminSearchForm.jsx

```diff
@@ -0,0 +1,163 @@
+    this.state = {
+      searchQuery: queryParams.search,
+      searchCourseQuery: queryParams.search_course,
+      searchDateQuery: queryParams.search_start_date,
+    };
+  }
+
+  componentDidUpdate(prevProps) {
+    const { location } = this.props;
+    if (location.search !== prevProps.location.search) {
+      // eslint-disable-next-line camelcase
+      const { search, search_course, search_start_date } = qs.parse(location.search);
+      const {
+        search: prevSearch,
+        search_course: prevSearchCourse,
+        search_start_date: prevSearchDate,
+      } = qs.parse(prevProps.location.search);
+      if (search !== prevSearch || search_course !== prevSearchCourse ||
+        search_start_date !== prevSearchDate) {
+        this.handleSearch(search, search_course, search_start_date);
+      }
+    }
+  }
+
+  onCourseSelect(event) {
+    const updateParams = {
+      search_course: event.target.value,
+      page: 1,
+    };
+    if (event.target.value === '') {
+      updateParams.search_start_date = '';
+    }
+    updateUrl(updateParams);
+  }
+
+  handleSearch(emailSearch, courseSearch, dateSearch) {
+    this.setState({
+      searchQuery: emailSearch,
+      searchCourseQuery: courseSearch,
+      searchDateQuery: dateSearch,
+    });
+    this.props.searchEnrollmentsList();
+  }
+
+  render() {
+    const { searchCourseQuery, searchDateQuery, searchQuery } = this.state;
+    const { tableData } = this.props;
+    const courseTitles = new Set(tableData.map(en => en.course_title));
+    const courseDates = new Set(tableData.map(en => en.course_start).sort().reverse());
+
+    return (
+      <div className="row">
+        <div className="col-12 pr-md-0 mb-0">
+          <div className="row w-100 m-0">
+            <div className="col-12 col-md-6 px-0">
+              <Form inline role="search">
+                <div className="col-6 pl-0 pr-md-2 pr-lg-3">
+                  <Form.Group>
+                    <Form.Label id="course-title-search" className="search-label mb-2">Filter by courses</Form.Label>
+                    <Form.Control
+                      className="w-100"
+                      as="select"
+                      aria-labelledby="course-title-search"
+                      value={searchCourseQuery}
+                      onChange={e => this.onCourseSelect(e)}
+                    >
+                      <option value="">All Courses</option>
+                      {courseTitles.map(title => (
+                        <option
+                          value={title}
+                          key={title}
+                        >
+                          {title}
+                        </option>))}
+                    </Form.Control>
+                  </Form.Group>
+                </div>
+                <div className="col-6 pr-0 px-md-2 px-lg-3">
+                  <Form.Group>
+                    <Form.Label id="date-search" className="search-label mb-2">Filter by start date</Form.Label>
```

Comment:
Did we decide to not do the help icon as per the miro spec, next to this label?

Reply:
We didn't have the text yet, I opened a separate ticket for it: https://openedx.atlassian.net/browse/ENT-3235